packages/core/src/components/index.ts

```diff
@@ -5,7 +5,12 @@
  * and https://github.com/palantir/blueprint/blob/master/PATENTS
  */
 
-import "dom4";
+declare function require(moduleName: string): any; // declare node.js "require" so that we can conditionally import
```

Comment:
note for a follow-up change (doesn't have to be in this PR): we should share this ambient declaration with https://github.com/palantir/blueprint/blob/497a03eab4420e48db0865407995895aa25726fa/packages/docs/src/require-shim.d.ts